bridges/TwitterBridge.php

```diff
@@ -234,7 +234,31 @@ public function collectData(){
 
 			// generate the title
 			$item['title'] = $tweet->full_text;
-			$cleanedTweet  = $tweet->full_text;
+
+			// Convert plain text URLs into HTML hyperlinks
+			if (isset($tweet->entities->media)) {
+				foreach($tweet->entities->media as $media) {
+					$cleanedTweet = str_replace($media->url,
+						'<a href="' . $media->expanded_url . '">' . $media->display_url . '</a>',
+						$tweet->full_text);
+				}
+			} elseif (isset($tweet->entities->urls)) {
+				foreach($tweet->entities->urls as $url) {
+					$cleanedTweet = str_replace($url->url,
+						'<a href="' . $url->expanded_url . '">' . $url->display_url . '</a>',
+						$tweet->full_text);
```

Comment:
If tweet has more than one url, only last url link will be converted. I suggest to join $tweet->entities->urls and and $tweet->entities->media using [array_merge](https://www.php.net/manual/ru/function.array-merge.php)

Reply:
array_merge will consume twice as much memory and I still have to check if objects exit first. I guess running two loops is probably as easy and faster.

Reply:
Fair enough